enderio-base/src/main/java/com/enderio/base/common/recipe/IEnderRecipe.java

```diff
@@ -0,0 +1,71 @@
+package com.enderio.base.common.recipe;
+
+import com.enderio.base.EnderIO;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.Container;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Recipe;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public interface IEnderRecipe<R extends IEnderRecipe<R, C>, C extends Container> extends Recipe<C> {
+    @Override
+    DataGenSerializer<R, C> getSerializer();
+
+    @Override
+    default boolean canCraftInDimensions(int width, int height) {
+        return true;
+    }
+
+    default String getOwningMod() {
+        return EnderIO.MODID;
```

Comment:
EnderIO is already filtered, maybe filter the namespace of the recipeid? So if the recipe is defined in machines it returns machines, so a "is machines present" check is not needed, even if the recipetype is belonging to Base or something

Reply:
So maybe thinking something like so?

```suggestion
        return Registry.RECIPE_TYPE.getKey(getType()).getNamespace();
```

Reply:
more like getID().getNamespace()

Reply:
I didn't see getID lmaoo, thanks